Regression tests and coverage setup

rathrio opened the pull request:
Adds some sanity tests to help detect regression and sets up coverage reporting.

More tests will be added incrementally.

rathrio committed:
Add InMemorySampleStorage.findBy for testing purposes

rathrio committed:
Add test for non primitive string representation

rathrio committed:
Add JS methods example

rathrio committed:
Install TruffleRuby on Travis

rathrio committed:
Use actual path to gu

rathrio committed:
Only build master branch

PRs will still be built as a different job.

rathrio committed:
Introduce Search for cleaner storage querying

Primarily useful for testing purposes

rathrio committed:
Don't access HashMap directly in tests

rathrio committed:
Consistent test naming and one test for Ruby

rathrio committed:
Fix root node id query for Ruby in tests

rathrio committed:
Add test for instrumenting R

rathrio committed:
R is weird

rathrio committed:
More ruby tests

rathrio committed:
Add test for ruby module function behaviour

rathrio committed:
Test for return values in module_function Ruby test

rathrio committed:
Configure jacoco

rathrio committed:
Add coverage report badge to README

rathrio committed:
Add Search usage doc

boris-spas commented on src/test/java/AudreyTest.java:
```diff
@@ -10,11 +10,10 @@
 import java.nio.file.Paths;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
It's usually bad practice to import *

boris-spas commented on src/test/java/AudreyTest.java:
```diff
@@ -34,10 +34,13 @@ public void setupAudrey() {
             .option("audrey", "true")
             .option("audrey.Project", "tests")
             .option("audrey.Storage", "in_memory")
+            .allowAllAccess(true)
```
Is this necessary in the end? If not it's safer to not have it so that "rm -rf /" cannot accidentally end up in tests.

rathrio replied:
R needs access to the native interface and wants to access some files in `/var/`. I made this [a little more restrictive now](https://github.com/rathrio/audrey/pull/6/commits/99eb5fef05d944210d0928f2d33faa6dbdd8f8f0), but I assume ["unrestricted IO operations"](https://github.com/oracle/graal/blob/5ff01c027cf59e4af379b775d38c42ce720ad202/sdk/src/org.graalvm.polyglot/src/org/graalvm/polyglot/Context.java#L936-L938) would allow `rm -rm /`?

Is there a more granular way to restrict file system access?

boris-spas commented on src/test/test_sources/add.R:
```diff
@@ -0,0 +1,6 @@
+    x + y
+}
+
+add(1, 2)
```
add newlines at end of files wherever it's missing.

rathrio committed:
Always use single class imports

For some reason IntelliJ only respects this when setting the
threshold to some large value.

https://stackoverflow.com/a/25916603

rathrio committed:
Be more explicit about accessing OS

R wants unrestrictive IO access + use the native interface

rathrio committed:
Add newlines at end of test files
